api/src/main/java/io/opentelemetry/trace/TraceId.java

```diff
@@ -85,50 +61,40 @@ public static TraceId getInvalid() {
    * @param random the random number generator.
    * @return a new valid {@code TraceId}.
    */
-  static TraceId generateRandomId(Random random) {
+  static CharSequence generateRandomId(Random random) {
     long idHi;
     long idLo;
     do {
       idHi = random.nextLong();
       idLo = random.nextLong();
     } while (idHi == INVALID_ID && idLo == INVALID_ID);
-    return new TraceId(idHi, idLo);
-  }
 
-  /**
-   * Returns a {@code TraceId} whose representation is copied from the {@code src} beginning at the
-   * {@code srcOffset} offset.
-   *
-   * @param src the buffer where the representation of the {@code TraceId} is copied.
-   * @param srcOffset the offset in the buffer where the representation of the {@code TraceId}
-   *     begins.
-   * @return a {@code TraceId} whose representation is copied from the buffer.
-   * @throws NullPointerException if {@code src} is null.
-   * @throws IndexOutOfBoundsException if {@code srcOffset+TraceId.getSize()} is greater than {@code
-   *     src.length}.
-   * @since 0.1.0
-   */
-  public static TraceId fromBytes(byte[] src, int srcOffset) {
-    Utils.checkNotNull(src, "src");
-    return new TraceId(
-        BigendianEncoding.longFromByteArray(src, srcOffset),
-        BigendianEncoding.longFromByteArray(src, srcOffset + BigendianEncoding.LONG_BYTES));
+    byte[] result = new byte[16];
+    BigendianEncoding.longToByteArray(idHi, result, 0);
```

Comment:
Shouldn't this be `return fromLongs(idHi, idLo)`?

Reply:
yep, good catch! I missed that one in the clean up.